kpatch-build/create-diff-object.c

```diff
@@ -771,6 +771,88 @@ static bool kpatch_line_macro_change_only(struct section *sec)
 
 	return true;
 }
+
+#elif __s390x__
+
+static int _insn_get_length(unsigned char *inst)
+{
+	int length;
+
+	switch (inst[0] >> 6) {
+		case 0b00:
+			length = 2;
+			break;
+		case 0b01:
+		case 0b10:
+			length = 4;
+			break;
+		case 0b11:
+			length = 6;
+			break;
+	}
+
+	return length;
+}
+
+static bool kpatch_line_macro_change_only(struct section *sec)
+{
+	unsigned char *start1, *start2;
+	unsigned long size, offset, length;
+	struct rela *rela;
+	int lineonly = 0, found;
+
+	if (sec->status != CHANGED ||
+	    is_rela_section(sec) ||
+	    !is_text_section(sec) ||
+	    sec->sh.sh_size != sec->twin->sh.sh_size ||
+	    !sec->rela ||
+	    sec->rela->status != SAME)
+		return false;
+
+	start1 = sec->twin->data->d_buf;
+	start2 = sec->data->d_buf;
+	size = sec->sh.sh_size;
+	for (offset = 0; offset < size; offset += length) {
+		length = _insn_get_length(start1 + offset);
+
+		if (_insn_get_length(start1 + offset) != _insn_get_length(start2 + offset))
+			return false;
+
+		if (!memcmp(start1 + offset, start2 + offset, length))
+			continue;
+
+		if ((start1[offset] != 0xa7) || (start1[offset+1] != 0x49) ||
```

Comment:
I know the disassembly is in the commit msg, but a quick one line comment (like the x86_64 version) before the instruction comparison would be nice:  `/* verify lghi %r4, <line number> */`